docs/mp/restclient/09_rest-client.adoc

```diff
@@ -34,69 +51,149 @@ include::{common-deps-page-prefix-inc}[tag=maven-dependency]
-                                 .build(SomeResource.class);
-
-someResource.someMethod(apiModel);
+SomeResource greetResource = RestClientBuilder.newBuilder()
+                                 .baseUri(URI.create("http://localhost:8080/greet"))
+                                 .build(GreetRestClient.class);
+greetResource.getDefaultMessage();
+----
+The `RestClientBuilder` interface extends the `Configurable` interface from Jakarta REST (JAX-RS),
+enabling direct registration of _providers_ such as filters, param converters, exception mappers, etc.
+Example:
+[source,java]
+----
+SomeResource greetResource = RestClientBuilder.newBuilder()
+                                 .baseUri(URI.create("http://localhost:8080"))
+                                 .register(GreetClientRequestFilter.class)
+                                 .register(GreetClientExceptionMapper.class)
+                                 .build(GreetRestClient.class);
+greetResource.getDefaultMessage();
 ----
 
-== Creating a New Client - CDI
+=== Creating a New Client Using CDI
 A rest client interface can be annotated with `@RegisterRestClient` to automatically register it with CDI.
-The `RegisterRestClient` annotation has a property `baseUri` that can be used to define the base endpoint of this client.
-This value can be overridden using configuration.
-
+This annotation has a property called `baseUri` that can be used to define the base endpoint to be
+used by the client to access the service.
 Example:
 [source,java]
 ----
-@RegisterRestClient(baseUri="http://localhost:8080/baseUri")
-public interface SomeResource {
-
-// ...
-
+@Path("/greet")
+@RegisterRestClient(baseUri="http://localhost:8080")
+public interface GreetRestClient {
+    // ...
 }
 ----
-
-Once a rest client interface is annotated, it can be injected into any CDI bean.
-
+Any Jakarta REST (JAX-RS) providers for a client can be registered using the (repeatable)
+`@RegisterProvider` annotation on the interface as shown below.
+Example:
+[source,java]
+----
+@Path("/greet")
+@RegisterRestClient(baseUri="http://localhost:8080")
+@RegisterProvider(GreetClientRequestFilter.class)
+@RegisterProvider(GreetClientExceptionMapper.class)
+public interface GreetRestClient {
+    // ...
+}
+----
+Once a rest client interface is annotated, it can be injected into any CDI bean. All properties
+in annotation `RegisterRestClient` can be overridden via configuration as described in
+<<Configuration options, Configuration options>>
 Example:
 [source,java]
 ----
 @Inject
 @RestClient
-SomeResource client;
-
+GreetRestClient client;
 // ...
+client.getMessage("Helidon");
+----
 
-client.sampleMethod();
+== Configuration
+rest client configuration is mainly used for CDI created and managed rest client instances. It will not work for those created programmatic way. Programmatically created instances are configured via methods present on `org.eclipse.microprofile.rest.client.RestClientBuilder`.
```

Comment:
Rest client configuration is ....
[Rest Client configuration is ...]
Applies to the entire doc: Is the official name of this feature the Helidon MicroProfile Rest Client or do we refer to the generic 'rest client' after introducing it? Is it a proper Helidon feature or is it part of the MP spec that we use?